subprojects/file-watching/src/main/java/org/gradle/internal/watch/vfs/impl/WatchingVirtualFileSystem.java

```diff
@@ -199,6 +207,10 @@ public FileSystemWatchingStatistics getStatistics() {
                             return fileSystemWatchingStatistics;
                         }
                     });
+                    if (verboseLogging) {
+                        logVerboseVfsStatistics(newRoot, "retains", "until next build");
```

Comment:
Why don't we use `fileSystemWatchingStatistics`, which already has all the information available?

Reply:
Ah. I reverted the deleted code from old, didn't realize we have a better way now. Will adapt.

Reply:
See https://github.com/gradle/gradle/pull/14342.